servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -603,22 +675,21 @@ private boolean deleteInodeInternal(Inode inode, boolean recursive, long opTimeM
     for (int i = delInodes.size() - 1; i >= 0; i --) {
       Inode delInode = delInodes.get(i);
 
-      if (delInode.isFile()) {
-        // Delete the ufs checkpoint.
-        String checkpointPath = ((InodeFile) delInode).getUfsPath();
-        if (!checkpointPath.isEmpty()) {
-          UnderFileSystem ufs = UnderFileSystem.get(checkpointPath, MasterContext.getConf());
-          try {
-            if (!ufs.exists(checkpointPath)) {
-              LOG.warn("File does not exist the underfs: " + checkpointPath);
-            } else if (!ufs.delete(checkpointPath, true)) {
-              return false;
-            }
-          } catch (IOException e) {
-            throw new TachyonException(e.getMessage());
-          }
+      // TODO(jiri): What should the Tachyon behavior be when a UFS delete operation fails?
```

Comment:
Oh okay, because its just a recursive delete we have no way of knowing which files were deleted. In that case maybe we just do a lazy sync later when the user tries to access the Tachyon file, we update the deleted status?